evaporate.js

```diff
@@ -885,7 +886,7 @@
                 }
                 var completedAt = new Date(u.completedAt || FAR_FUTURE);
 
-                return con.partSize === u.partSize && completedAt > HOURS_AGO;
+                return con.partSize === u.partSize && completedAt > HOURS_AGO && me.name === u.awsKey && con.bucket === u.bucket;
```

Comment:
@cnorthwood This looks good but the `me.name === uwsKey` breaks existing functionality. Checking for the bucket is definitely something that needs to be added, though.

I think that EvaporateJS already gives you what you need in it's config options `allowS3ExistenceOptimization` (defaults to `false`).